examples/src/main/java/com/apple/foundationdb/record/sample/Main.java

```diff
@@ -127,6 +128,7 @@ public static void main(String[] args) {
         RecordMetaData rmd = RecordMetaData.build(SampleProto.getDescriptor());
 
         FDBRecordStore.Builder recordStoreBuilder = FDBRecordStore.newBuilder()
+                .setContext(context)
```

Comment:
Hm, I'm not sure what to do about this. The idea with #151 was that you could have a single key space path that you reused for multiple record stores. Then you could create a record store builder *without* a context but *with* a key space path and use that at store construction time.

I filed an issue to make it so that the order of calling `setContext` and `setKeySpacePath` can be made in any order: #338. In the mean time, I think the better course of action might be to have a `getRecordStore` method defined in this class. It could do something like take an `FDBRecordContext` and then produce an `FDBRecordStore` with the meta-data and path set (I don't know) from member variables (in production you'd probably make it part of some other kind of state object) and the context from its parameter? Or something like that.

Reply:
I don't know...I think I'm OK with this change as it stands. We have effectively deprecated the `KeySpacePath` carrying around its own `FDBRecordContext` so we really shouldn't be showing that usage in a test program anyway and, when issue #169 is done this will be an illegal operation anyway.

Reply:
Oh, to be clear, the code wasn't using the deprecated API--it was just throwing an error whenever it was run.

Reply:
thanks @alecgrieser @scgray for review and comments. Is it acceptable to have current codes to merge in before #169 is done? After #169, we can consider how to adapt without `setKeySpacePath()`?